src/user/user.controller.ts

```diff
@@ -26,16 +29,31 @@ export class UserController {
     }
   }
 
+  @Get('profile')
+  @UseGuards(JwtAuthGuard)
+  async getProfile(@Req() req: Request & { user: any }) {
+    if (!req.user || !req.body.email) {
+      throw new BadRequestException('Invalid user data.');
+    }
+    return this.userService.findByEmail(req.body.email);
+  }
+  
+
+@Get(':id')
+findOne(@Param('id') id: string) {
+  const numericId = Number(id);
+  if (isNaN(numericId) || numericId <= 0) {
```

Comment:
You can use DTOs to automatically validate [route params](https://docs.nestjs.com/pipes#built-in-pipes) instead of doing it manually

Reply:
fixed by https://github.com/alkemyTech/CENCO-MIT-Back-T1/pull/21/commits/f9f03caa56135c9e5e686147c7d439fe5d342767